service/history/deletemanager/delete_manager.go

```diff
@@ -210,15 +211,17 @@ func (m *DeleteManagerImpl) deleteWorkflowExecutionInternal(
 	// TODO (alex): Remove them when cassandra standard visibility is removed.
 	var startTime *time.Time
 	var closeTime *time.Time
-	// There are cases when workflow execution is closed but visibility is not updated and still open.
-	// This happens, for example, when workflow execution is deleted right from CloseExecutionTask.
-	// Therefore, force to delete from open visibility regardless of execution state.
-	if forceDeleteFromOpenVisibility || ms.GetExecutionState().State != enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED {
-		startTime = ms.GetExecutionInfo().GetStartTime()
-	} else {
-		closeTime, err = ms.GetWorkflowCloseTime(ctx)
-		if err != nil {
-			return err
+	if m.visibilityManager.HasStoreName(cassandra.CassandraPersistenceName) {
```

Comment:
When these values are nil, the visibility task will fail (panic) when using Cassandra.